LanguageExt.CodeGen/CodeGenUtil.cs

```diff
@@ -38,16 +38,38 @@ public static (ClassDeclarationSyntax PartialClass, TypeSyntax ReturnType, List<
-                                     .Select(m => m as FieldDeclarationSyntax)
-                                     .Where(f => f.Declaration.Variables.Count > 0)
-                                     .Where(f => FirstCharIsUpper(f.Declaration.Variables[0].Identifier.ToString()))
-                                     .Where(f => f.Modifiers.Any(SyntaxKind.PublicKeyword))
-                                     .Where(f => f.Modifiers.Any(SyntaxKind.ReadOnlyKeyword))
-                                     .Where(f => !f.Modifiers.Any(SyntaxKind.StaticKeyword))
-                                     .ToList();
-
+            var variables = applyToClass.Members
+                                        .Where(m => m is FieldDeclarationSyntax)
+                                        .Select(m => m as FieldDeclarationSyntax)
+                                        .Where(f => f.Declaration.Variables.Count > 0)
+                                        .Where(f => FirstCharIsUpper(f.Declaration.Variables[0].Identifier.ToString()))
+                                        .Where(f => f.Modifiers.Any(SyntaxKind.PublicKeyword))
+                                        .Where(f => f.Modifiers.Any(SyntaxKind.ReadOnlyKeyword))
+                                        .Where(f => !f.Modifiers.Any(SyntaxKind.StaticKeyword))
+                                        .Select(f => ( 
+                                            f.Declaration.Variables[0].Identifier,
+                                            f.Declaration.Type,
+                                            f.Modifiers
+                                        ));
+   
+            var properties = applyToClass.Members
+                                         .Where(m => m is PropertyDeclarationSyntax)
+                                         .Select(m => m as PropertyDeclarationSyntax)
+                                         .Where(p => FirstCharIsUpper(p.Identifier.ToString()))
+                                         .Where(p => p.Modifiers.Any(SyntaxKind.PublicKeyword))
+                                         .Where(p => !p.Modifiers.Any(SyntaxKind.StaticKeyword))
+                                         .Where(p => p.AccessorList.Accessors.Count == 1)
+                                         .Where(p => p.AccessorList.Accessors[0].Kind() == SyntaxKind.GetAccessorDeclaration)
+                                         .Where(p => p.AccessorList.Accessors[0].ExpressionBody == null)
+                                         .Where(p => p.AccessorList.Accessors[0].Body == null)
+                                         .Where(p => p.Initializer == null)
+                                         .Select(p => (
+                                             p.Identifier,
+                                             p.Type,
+                                             p.Modifiers
+                                         ));
+ 
+            var fields = variables.Concat(properties).ToList();
```

Comment:
This loses the ordering of fields and properties inthe class.  The guidance is that the order the fields are in the class are the order of the entries to the constructor.  Anyone mixing the order of the fields and properties would be confused at this.

One technique might be to zip the members with an `Enumerable.Range(0, Int32.MaxValue)` to create an index for each member and then use that index to sort the final set of results. 

It'd be good to rename `variables`  to fields and `fields` to `members` also.

Reply:
Done. A bit redundant to impose restrictions on both naming and ordering of the constructor argument, don't you think? But that's out of topic here.